apps/core/app/[locale]/(default)/blog/[blogId]/_components/sharing-links.tsx

```diff
@@ -3,22 +3,32 @@
 import { SiFacebook, SiLinkedin, SiPinterest, SiX } from '@icons-pack/react-simple-icons';
 import { Mail, Printer } from 'lucide-react';
 
-interface SharingLinksProps {
-  blogPostId: string;
-  blogPostImageUrl?: string;
-  blogPostTitle?: string;
+import { FragmentOf, graphql } from '~/client/graphql';
+
+import { SEOFragment } from '../page-data';
+
+export const SharingLinksFragment = graphql(
+  `
+    fragment SharingLinksFragment on BlogPost {
+      entityId
+      thumbnailImage {
+        altText
+        url: urlTemplate
+      }
+      ...SEOFragment
+    }
+  `,
+  [SEOFragment],
+);
+
+interface Props {
+  blogPost: FragmentOf<typeof SharingLinksFragment>;
   vanityUrl?: string;
```

Comment:
I'm curious what is the best approach with this field. [Here](https://github.com/bigcommerce/catalyst/pull/848/files#diff-e2a5797679e0fca50a55870caa3f97d1bde21078f613ddf55a3722951627fa02R44) we return as the result of `getBlogPost` but it's not a field in a BlogPost fragment, it is a setting in Blog.